packages/jest-matchers/src/__tests__/matchers-test.js

```diff
@@ -455,3 +455,61 @@ describe('.toMatch()', () => {
     jestExpect('this?: throws').toMatch('this?: throws');
   });
 });
+
+describe('toMatchObject()', () => {
+
+  [
+    [{a: 'b', c: 'd'}, {a: 'b'}],
+    [{a: 'b', c: 'd'}, {a: 'b', c: 'd'}],
+    [{a: 'b', t: {z: 'z', x: {r: 'r'}}}, {a: 'b', t: {z: 'z'}}],
+    [{a: 'b', t: {z: 'z', x: {r: 'r'}}}, {t: {x: {r: 'r'}}}],
+    [{a: [3, 4, 5], b: 'b'}, {a: [3, 4, 5]}],
+    [{a: [3, 4, 5], b: 'b'}, {a: [3, 4]}],
```

Comment:
I agree with you! I have removed this behaviour and have added a check that an equality check is performed for the elements in the array.

So this test will now fail:

```js
expect({permissions: ['read', 'write'], ...manyOtherProps}).toMatchObject({permissions: ['read']});
```

There is one exception with the equality check mentioned above; you can still perform matchObject checks for key-value objects contained in arrays (i.e. array of objects), so in my update this will pass: 

```js
expect([{a: 'a', b: 'b'}, 5]).toMatchObject([{a: 'a'}, 5])
```

Do you agree with this behaviour?